### DISCUSSION TITLE
Use with generic XMPP server

### DISCUSSION THREAD

**Discussion by guusdk** [prose-im/prose-app-web#112] [General]
Hi there!

Is the Prose web app thought to be suitable for use with a generic XMPP server, or will the client require a custom server?

As one of the Openfire developers, I've spent some time creating a simple plugin that wraps your client, for easy deployment in our server. I'm running into some issues, and am considering to raise them here. However, if the client is not _supposed_ to work with other server implementations, then there's little point in doing so.

Keep up the good work!

**Comment by valeriansaliou** [comment]
Hey @guusdk !

Since Prose was designed with workspace uses in mind, it requires some custom configuration rules on the server, which is typically closed to federation for safety (due to phishing). All servers members are already in all other users rosters, so that you don't have to manually add all your team's contacts.

It's quite protocol-standard other than that, we never tested it on Openfire, but you may review our reference Prosody configuration file here: https://github.com/prose-im/prose-pod-system/blob/3b1e897a8b563c7320bdc4e6a55a2a88e9243959/server/etc/prosody/prosody.cfg.lua

Also, I want to be transparent that since Prose's business model will be based on a future cloud hosting offering (that would make it a sustainable OSS project that can fund itself on its own - for now it comes from my own money), and for branding reasons, I'd request that any easy integration into XMPP server plugin offering do not rebrand the app, and make explicit references to the Prose project. Other than than, it's fine! :)

PS: Prose is still a WIP and is still at the alpha stage, getting closer to beta. We'll start writing extensive docs once we get closer to a RC.

**Reply by guusdk** [reply]
Thanks for the quick response @valeriansaliou. That's all very reasonable, thank you.

We would certainly not re-brand the app, if we ever get to a stage where we release a plugin (for reference, have a look at [some](https://www.igniterealtime.org/projects/openfire/plugin-archive.jsp?plugin=xmppweb) of the [other](https://www.igniterealtime.org/projects/openfire/plugin-archive.jsp?plugin=jsxc) web [clients](https://www.igniterealtime.org/projects/openfire/plugin-archive.jsp?plugin=inverse) for which we provide [wrapper](https://www.igniterealtime.org/projects/openfire/plugin-archive.jsp?plugin=candy) plugins. We're explicitly using the original project name to avoid confusion).

Given that you're not _opposed_ us wrapping the client, I'll start raising issues in this repository for things that get in the way of that. [Here's the first one.](https://github.com/prose-im/prose-app-web/issues/113) There's obviously no expectation that you commit to resolving these.

**Reply by valeriansaliou** [reply]
Excellent! Thanks. I'll confirm per-issue if they can get processed (at any time in the future) or not so that you have some view on that.

**Comment by poVoq** [comment]
It seems to work fine with Ejabberd from the limited testing I have done.
